cpp/ql/test/library-tests/ir/types/irtypes.cpp

```diff
@@ -0,0 +1,65 @@
+struct A {
+  int f_a;
+};
+
+struct B {
+  double f_a;
+  float f_b;
+};
+
+enum E {
+  Zero,
+  One,
+  Two,
+  Three
+};
+
+enum class ScopedE {
+  Zero,
+  One,
+  Two,
+  Three
+};
+
+void IRTypes() {
+  char c;  //$irtype=int1
+  signed char sc;  //$irtype=int1
+  unsigned char uc;  //$irtype=uint1
+  short s;  //$irtype=int2
+  signed short ss;  //$irtype=int2
+  unsigned short us;  //$irtype=uint2
+  int i;  //$irtype=int4
+  signed int si;  //$irtype=int4
+  unsigned int ui;  //$irtype=uint4
+  long l;  //$irtype=int8
+  signed long sl;  //$irtype=int8
+  unsigned long ul;  //$irtype=uint8
+  long long ll;  //$irtype=int8
+  signed long long sll;  //$irtype=int8
+  unsigned long long ull;  //$irtype=uint8
+  bool b;  //$irtype=bool1
+  float f;  //$irtype=float4
+  double d;  //$irtype=float8
+  long double ld;  //$irtype=float16
+  __float128 f128;  //$irtype=float16
```

Comment:
Why no complex or imaginary types in this test?

Reply:
They're in `complex.c`, since they seem to be supported only in C99 mode.